### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by jturkel**
I borrowed pretty heavily from the [omniauth-multi-provider-saml ](https://github.com/salsify/omniauth-multi-provider-saml) README :)

@tjwp - you're prime

### CODE REVIEW

**Comment by tjwp** [comment]
Looks like you either need to add a `.ruby-version` or define TargetRubyVersion for rubocop.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
-Welcome to your new gem! In this directory, you'll find the files you need to be
-able to package up your Ruby library into a gem. Put your Ruby code in the file
-`lib/omniauth-multi-provider`. To experiment with that code, run 
-`bin/console` for an interactive prompt.
-
-TODO: Delete this and the text above, and describe your gem
+This is a simple extension to [omniauth](https://github.com/omniauth/omniauth) for supporting 
+multiple identity provider instances of a given type e.g. multiple SAML 
+identity providers. It is a generalization of the 
+[omniauth-multi-provider-saml](https://github.com/salsify/omniauth-multi-provider-saml).
```
Comment (tjwp) [inline README.md]: Is there going to be notice on this repo that the gem is deprecated and this generalization is being maintained instead?

File: README.md [hunk 2]
```diff
@@ -23,9 +21,61 @@ Or install it yourself as:
 
     $ gem install omniauth-multi-provider
 
-## Usage
+## Setup
+
+**I would highly recommend first getting your setup to work with a single identity provider before
```
Comment (tjwp) [inline README.md]: Should this be first person, or "It is highly recommended ..."?

File: README.md (hunk 1 above)
Comment (jturkel) [inline README.md]: Yup. Will do it after I tie off the README + tests.

**Comment by jturkel** [comment]
Updated to mention multiple provider types and include multiple provider types in the example config.

**Pull request merged**